hls-writer/hls_writer.py

```diff
@@ -225,22 +237,42 @@ def hls_writer(layer_list, yamlConfig):
+                 activation_name = layer_list[i-1]['activation']+'_config'+str(i)
+                 if layer_list[i-1]['activation'] == "relu":
+                     newline += '    nnet::relu<{}, {}, {}>(logits{}, {});\n'.format(output_type, output_type, activation_name, i, output_object)
+                 elif layer_list[i-1]['activation'] =="softmax":
+                     newline += '    nnet::softmax<{}, {}, {}>(logits{}, {});\n'.format(output_type, output_type, activation_name, i, output_object)
+                 elif layer_list[i-1]['activation'] =="sigmoid":
+                     newline += '    nnet::sigmoid<{}, {}, {}>(logits{}, {});\n'.format(output_type, output_type, activation_name, i, output_object)
+                 elif layer_list[i-1]['activation'] =="tanh":
+                     newline += '    nnet::tanh<{}, {}, {}>(logits{}, {});\n'.format(output_type, output_type, activation_name, i, output_object)
+                 elif layer_list[i-1]['activation'] =="linear": 
+                     #github Issue 53
+                     newline += '    nnet::linear<{}, {}, {}>(logits{}, {});\n'.format(output_type, output_type, activation_name, i, output_object)
+                 else:
+                     raise Exception('ERROR: MISSING ACTIVATION')
+
+                if layer_list[i-1]['class_name'] == 'Activation':
+                 activation_name = layer_list[i-1]['activation']+'_config'+str(i)
+                 if layer_list[i-1]['activation'] == "relu":
+                    newline += '    nnet::relu<{}, {}, {}>({}, {});\n'.format(input_type, output_type, activation_name, input_object, output_object)
+                 elif layer_list[i-1]['activation'] =="softmax":
+                    newline += '    nnet::softmax<{}, {}, {}>({}, {});\n'.format(input_type, output_type, activation_name, input_object, output_object)
+                 elif layer_list[i-1]['activation'] =="sigmoid":
+                    newline += '    nnet::sigmoid<{}, {}, {}>({}, {});\n'.format(input_type, output_type, activation_name, input_object, output_object)
+                 elif layer_list[i-1]['activation'] =="tanh":
+                    newline += '    nnet::tanh<{}, {}, {}>({}, {});\n'.format(input_type, output_type, activation_name, input_object, output_object)
+                 elif layer_list[i-1]['activation'] == "binary_tanh":	
+                    newline += '    nnet::binary_tanh<{}, {}, {}>({}, {});\n'.format(input_type, output_type, activation_name, input_object, output_object)
+                 elif layer_list[i-1]['activation'] =="linear": 
                     #github Issue 53
-                    newline += '    nnet::linear<{}, {}, {}>(logits{}, {});\n'.format(output_type, output_type, activation_name, i, output_object)
-                else:
+                    newline += '    nnet::linear<{}, {}, {}>(logits{}, {});\n'.format(input_type, output_type, activation_name, input_object, output_object)
```

Comment:
Is the fact that this `linear` looks different from the one above related to what Activation is?